### DISCUSSION TITLE
Saving data from block to custom table

### DISCUSSION THREAD

**Discussion by Lovor01** [WordPress/gutenberg#65037] [Q&A]
I want to have additional functionality of my custom block to save some metadata (email) to custom table. The reason is to obfuscate this from post content and make it available to backend when frontend requests sending mail via id key.

I attempted this: I created custom table, created entity and used useEntityRecord to read and save data from table, created custom REST endpoints. So far so good, everything works.

But, when email address is edited in the block, editor shows second save option (like when saving from site editor and you change both template and page) - because email is separate entity:

```js
dispatch( store ).addEntities( [
	{
		label: __( 'Save form settings from admin' ),
		name: 'mie-save-email',
		kind: 'postType',
		baseURL: 'mie-save-email/v1/email',
		baseURLParams: { context: 'edit' },
	},
] );
```
Ideally, email would be saved to separate table when saving post, but in the editor there would be no question about saving two separate things (block also has attributes that can be changed), saving should function like normal blocks do.

**Comment by AlphaZebra** [comment]
This is a little above my paygrade, but I'll just mention that I write to and from custom tables all the time in some blocks I've written, and I just treat them as variables in an array (that is, as returned from a DB query). The block editor has no idea of my shenanigans and there's no second entity... Is that a direction to explore, or is the entity part of this needed?

**Reply by Lovor01** [reply]
Entity is needed to synchronize saving with saving of the block and ease sending data to REST. Mind you that's happening in edit function in JavaScript, not in PHP. I didn't quite catch how you do it, it seemed to me that you are talking about saving in PHP rendering part

**Reply by AlphaZebra** [reply]
I see, yes, why you're using an entity. The stuff I've done has been front end, that's true. But you could make the REST call to save the custom data in the save.js file... would that work?

**Reply by Lovor01** [reply]
I appreciate your effort, but no, that would not work. Save function must be a pure function without side effects. This must be done in edit.js and should be synchronized with saving in editor. I could attempt catching isSavingPost and isAutoSavingPost actions in subscribe to store and then write to database, but that feels like a hack.
